storagedriver/s3/s3.go

```diff
@@ -580,6 +580,16 @@ func (d *Driver) Delete(path string) error {
 	return nil
 }
 
+// URLFor returns a URL which may be used to retrieve the content stored at the given path.
+// May return an UnsupportedMethodErr in certain StorageDriver implementations.
+func (d *Driver) URLFor(path string) (string, error) {
+	if !storagedriver.PathRegexp.MatchString(path) {
+		return "", storagedriver.InvalidPathError{Path: path}
+	}
+
+	return d.Bucket.SignedURL(d.s3Path(path), time.Now().Add(24*time.Hour)), nil
```

Comment:
For whatever reason the old registry created S3 URLs which were valid for 20 minutes ([reference](https://github.com/docker/docker-registry/blob/master/docker_registry/drivers/s3.py#L155)). I'm not sure what the recommended time frame should be, but I feel like 1 day is probably excessive.